src/main/java/net/earthcomputer/clientcommands/command/MinesweeperCommand.java

```diff
@@ -0,0 +1,410 @@
+                for (int y = 0; y < boardHeight; y++) {
+                    boolean hovered = Mth.floorDiv(mouseX - topLeftX - 12, 16) == x && Mth.floorDiv(mouseY - topLeftY - 12, 16) == y;
+                    blitSprite(graphics, getTileSprite(x, y, hovered),  x * 16 + 12, y * 16 + 12, 16, 16);
+                }
+            }
+        }
+
+        public void blitSprite(GuiGraphics graphics, Vector2i uv, int x, int y, int width, int height) {
+            graphics.blit(MINESWEEPER_ATLAS, topLeftX + x, topLeftY + y, width, height, uv.x, uv.y, width, height, MINESWEEPER_ATLAS_WIDTH, MINESWEEPER_ATLAS_HEIGHT);
+        }
+
+        @Override
+        public void tick() {
+            if (ticksPlaying > 0 && gameActive()) {
+                ticksPlaying += 1;
+            }
+        }
+
+        @Override
+        public boolean mouseReleased(double mouseX, double mouseY, int button) {
+            int tileX = Mth.floorDiv((int) (mouseX - topLeftX - 12), 16);
+            int tileY = Mth.floorDiv((int) (mouseY - topLeftY - 12), 16);
+
+            if (isWithinBounds(tileX, tileY) && gameActive()) {
+                if (button == InputConstants.MOUSE_BUTTON_LEFT) {
+                    if (ticksPlaying == 0) {
+                        generateMines(tileX, tileY);
+                        ticksPlaying = 1;
+                    }
+
+                    click(tileX, tileY);
+
+                    assert minecraft != null && minecraft.player != null;
+                    if (emptyTilesRemaining <= 0) {
+                        minecraft.player.playNotifySound(SoundEvents.NOTE_BLOCK_PLING.value(), SoundSource.MASTER, 1.0f, 2.0f);
+                    } else if (deathCoords != null) {
+                        minecraft.player.playNotifySound(SoundEvents.NOTE_BLOCK_BASS.value(), SoundSource.MASTER, 1.0f, 1.0f);
+                    }
+                } else if (button == InputConstants.MOUSE_BUTTON_RIGHT) {
+                    flag(tileX, tileY);
+                }
+            }
+
+            return true;
+        }
+
+        private boolean gameActive() {
+            return deathCoords == null && emptyTilesRemaining > 0;
+        }
+
+        private void generateMines(int avoidX, int avoidY) {
+            for (int i = 0; i < mines; i++) {
+                int x = random.nextInt(boardWidth);
+                int y = random.nextInt(boardHeight);
+
+                // too close to the clicked position
+                if (Mth.abs(avoidX - x) <= 1 && Mth.abs(avoidY - y) <= 1) {
+                    i--;
+                    continue;
+                }
+
+                if (tileType(getTile(x, y)) == MINE_TILE_TYPE) {
+                    i--;
+                    continue;
+                }
+
+                incrementWarning(x - 1, y - 1);
+                incrementWarning(x, y - 1);
+                incrementWarning(x + 1, y - 1);
+
+                incrementWarning(x - 1, y);
+                board[y * boardWidth + x] = 0b000_10_0_0;
```

Comment:
This line needs a comment, but preferably I'd like to see this magic value be expressed in terms of the named constant you have for the mine type already. For instance, `MINE_TILE_TYPE << TILE_TYPE_BIT`

Reply:
I’ll make a create tile function don't worry